book/src/dev/rfcs/0005-state-updates.md

```diff
@@ -61,7 +62,64 @@ state service.
 # Guide-level explanation
 [guide-level-explanation]: #guide-level-explanation
 
-XXX fill in after writing other details
+The `zebra-state` crate provides an implementation of the chain state storage
+logic in a zcash consensus node. Its main responsibility is to store chain
+state, validating new blocks against the existing chain state in the process,
+and to allow later querying of said chain state. `zebra-state` provides this
+interface via a `tower::Service` based on the actor model with a
+request/response interface for passing messages back and forth between the
+state service and the rest of the application.
+
+The main entry point for the `zebra-state` crate is the `init` function. This
+function takes a `zebra_state::Config` and constructs a new state service,
+which it returns wrapped by a `tower::Buffer`. This service is then interacted
+with via the `tower::Service` trait.
+
+```rust
+use tower::{Service, ServiceExt};
+
+let state = zebra_state::on_disk::init(state_config, network);
+let request = zebra_state::Request::BlockLocator;
+let response = state.ready_and().await?.call(request).await?;
+
+assert!(matches!(response, zebra_state::Response::BlockLocator(_)));
+```
```

Comment:
Ooh we should make this into a doctest in the state docs

Reply:
created an issue for this